test/test-tcp-bind-error.c

```diff
@@ -297,3 +297,28 @@ TEST_IMPL(tcp_bind_writable_flags) {
   MAKE_VALGRIND_HAPPY();
   return 0;
 }
+
+TEST_IMPL(tcp_bind_or_listen_error_after_close) {
+  uv_loop_t loop;
+  uv_tcp_t tcp;
+  struct sockaddr_in addr;
+
+  memset(&addr, 0, sizeof(addr));
+  addr.sin_addr.s_addr = htonl(INADDR_ANY);
+  addr.sin_port = htons(9999);
+  addr.sin_family = AF_INET;
+
+  int r;
+  r = uv_loop_init(&loop);
+  ASSERT(r == 0);
+  r = uv_tcp_init(&loop, &tcp);
+  ASSERT(r == 0);
+  uv_close((uv_handle_t *)&tcp, NULL);
+  r = uv_tcp_bind(&tcp, (struct sockaddr *)&addr, 0);
+  ASSERT(r == UV_EINVAL);
+  r = uv_listen((uv_stream_t *)&tcp, 5, NULL);
+  ASSERT(r == UV_EINVAL);
+  (void)uv_run(&loop, UV_RUN_DEFAULT);
```

Comment:
I don't think you need the `(void)` here, but you can assert on the return value of `uv_run()`.

Reply:
Done.